From: Kai Liu <kai.liu@suse.com>
Date: Wed, 1 Dec 2021 10:38:34 +0800
Subject: Fix integrity.h compilation error
Patch-mainline: Never, fix difference between openEuler and mainline code
openEuler-commit: 1d8a803c67564b5a298ff19b56b721a583e0e291
References: bsn#22

This patch fixes below compilation error:

  CC      security/integrity/evm/evm_secfs.o
In file included from security/integrity/evm/evm.h:18:0,
                 from security/integrity/evm/evm_secfs.c:17:
security/integrity/evm/../integrity.h:96:12: error: ‘SHA512_DIGEST_SIZE’ undeclared here (not in a function); did you mean ‘SHA1_DIGEST_SIZE’?
  u8 digest[SHA512_DIGEST_SIZE];
            ^~~~~~~~~~~~~~~~~~
            SHA1_DIGEST_SIZE
make[3]: *** [scripts/Makefile.build:280: security/integrity/evm/evm_secfs.o] Error 1
make[2]: *** [scripts/Makefile.build:497: security/integrity/evm] Error 2
make[1]: *** [scripts/Makefile.build:497: security/integrity] Error 2
make: *** [Makefile:1822: security] Error 2

openEuler ported mainline commit:
  a24d22b225ce crypto: sha - split sha.h into sha1.h and sha2.h
as 
  1d8a803c6756 crypto: sha - split sha.h into sha1.h and sha2.h

but they secretly changed "#include <crypto/sha1.h>" to sha2.h in integrity.h
without submitting a separate commit.

Signed-off-by: Kai Liu <kai.liu@suse.com>

---
 security/integrity/integrity.h |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/security/integrity/integrity.h
+++ b/security/integrity/integrity.h
@@ -14,7 +14,7 @@
 
 #include <linux/types.h>
 #include <linux/integrity.h>
-#include <crypto/sha1.h>
+#include <crypto/sha2.h>
 #include <linux/key.h>
 #include <linux/audit.h>
 #include <linux/hash_info.h>
